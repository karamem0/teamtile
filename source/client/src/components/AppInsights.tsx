//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import { ReactPlugin, withAITracking } from '@microsoft/applicationinsights-react-js';
import { ApplicationInsights } from '@microsoft/applicationinsights-web';

import { createBrowserHistory } from 'history';

import Presenter from './AppInsights.presenter';

const browserHistory = createBrowserHistory();
const reactPlugin = new ReactPlugin();

const instrumentationKey = process.env.VITE_APPLICATIONINSIGHTS_INSTRUMENTATION_KEY;
if (instrumentationKey) {
  const appInsights = new ApplicationInsights({
    config: {
      instrumentationKey: instrumentationKey,
      extensions: [ reactPlugin ],
      extensionConfig: {
        [reactPlugin.identifier]: { history: browserHistory }
      }
    }
  });
  appInsights.loadAppInsights();
}

export default withAITracking(reactPlugin, Presenter);
