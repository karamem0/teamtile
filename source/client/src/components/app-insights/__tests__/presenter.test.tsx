//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { render, screen } from '@testing-library/react';

import AppInsights from '../presenter';

beforeEach(() => {
  jest.clearAllMocks();
  jest.restoreAllMocks();
});

describe('CallbackPage', () => {

  it('create shapshot', () => {
    const params = {
      children: <React.Fragment />
    };
    render(<AppInsights {...params} />);
    expect(screen.queryAllByText(/^.*$/)[0]).toMatchSnapshot();
  });

});
