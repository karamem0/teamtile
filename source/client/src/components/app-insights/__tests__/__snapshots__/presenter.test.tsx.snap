// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallbackPage create shapshot 1`] = `
<body>
  <div />
</body>
`;
